container/sagemaker/proxy_client.py

```diff
@@ -64,13 +64,17 @@ def add_model(self, model_name, base_path, model_platform='tensorflow'):
                 req = model_management_pb2.ReloadConfigRequest()
                 req.config.CopyFrom(model_server_config)
 
-                self.stub.HandleReloadConfigRequest(req)
+                self.stub.HandleReloadConfigRequest(request=req,
+                                                    timeout=5,
+                                                    wait_for_ready=True)
                 self._add_model_to_config_file(model_name, base_path, model_platform)
             except grpc.RpcError as e:
                 if e.code() is grpc.StatusCode.INVALID_ARGUMENT:
                     raise Exception(409, e.details())
+                elif e.code() is grpc.StatusCode.DEADLINE_EXCEEDED:
+                    raise Exception(408, e.details())
                 else:
-                    raise Exception(e.code(), e.details())
+                    raise Exception(500, e.details())
```

Comment:
Are we sure we want all elses to throw 500?

Reply:
I'm encoding all other errors to 500 because the TFS server does not return a HTTP error code, instead, it returns the grpc error codes defined here: https://github.com/tensorflow/serving/blob/master/tensorflow_serving/apis/get_model_status.proto

Same as the comment in `python_service.py`

We can however parse each grpc error codes and map them to different http codes.

Reply:
Did you link the right file? Those look like status codes.

Reply:
You are right. I previously thought falcon response cannot propagate grpc status codes. Looks like it can. Updated.